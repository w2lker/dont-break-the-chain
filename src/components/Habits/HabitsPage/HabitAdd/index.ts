import HabitAdd from './HabitAdd.decorators';

export default HabitAdd;
