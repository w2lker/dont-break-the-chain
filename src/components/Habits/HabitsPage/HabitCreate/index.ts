import HabitCreate from './HabitCreate.decorators';

export default HabitCreate;
