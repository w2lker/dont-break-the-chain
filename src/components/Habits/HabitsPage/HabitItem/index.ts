import HabitItem from './HabitItem.decorators';

export default HabitItem;
