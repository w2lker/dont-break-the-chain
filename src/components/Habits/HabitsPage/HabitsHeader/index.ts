import HabitsHeader from './HabitsHeader.decorators';

export default HabitsHeader;
