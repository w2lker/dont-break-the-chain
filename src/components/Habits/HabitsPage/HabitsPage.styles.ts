const habitsPageStyles = {
  root: {
    padding: 0,
  },
};

export default habitsPageStyles;
