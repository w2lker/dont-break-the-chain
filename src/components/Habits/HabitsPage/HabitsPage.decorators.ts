import withStyles from '@material-ui/core/styles/withStyles';
import HabitsPage from './HabitsPage';
import habitsPageStyles from './HabitsPage.styles';

const styledComponent = withStyles(habitsPageStyles)(HabitsPage);

export default styledComponent;
