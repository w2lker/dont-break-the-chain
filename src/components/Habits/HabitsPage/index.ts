import HabitsPage from './HabitsPage.decorators';

export default HabitsPage;
