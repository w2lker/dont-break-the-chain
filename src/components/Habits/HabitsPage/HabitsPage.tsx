import React from 'react';

import { WithStyles } from '@material-ui/styles';

import HabitsHeader from './HabitsHeader';

import habitsPageStyles from './HabitsPage.styles';

export interface IHabitsPageProps extends WithStyles<typeof habitsPageStyles> {}

const HabitsPage: React.FC<IHabitsPageProps> = (props) => {
  const { classes } = props;
  return (
    <div className={classes.root}>
      <HabitsHeader />
    </div>
  );
};

export default HabitsPage;
