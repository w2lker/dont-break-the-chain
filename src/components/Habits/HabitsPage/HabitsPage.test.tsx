import React from 'react';

import { mount, shallow } from 'enzyme';
// import cases from 'jest-in-case';

import HabitsPage, { IHabitsPageProps } from './HabitsPage';
import HabitsPageDecorated from './HabitsPage.decorators';
import habitsPageStyles from './HabitsPage.styles';

const testSetup = () => {
  const classes = {
    root: 'habitsPage-wrapper',
  };
  const classesKeys = Object.keys(classes);
  const sampleProps: IHabitsPageProps = { classes };

  return { classes, classesKeys, sampleProps };
};

describe('HabitsPage component', () => {

  it('match snapshot', () => {
    const { sampleProps } = testSetup();
    const component = shallow(<HabitsPage {...sampleProps} />);
    expect(component.debug()).toMatchSnapshot();
  });

  it('Renders header', () => {
    const { sampleProps } = testSetup();
    const component = mount(<HabitsPage {...sampleProps} />);
    const header = component.find('HabitsHeader');
    expect(header.length).toBe(1);
  });
});

describe('HabitsPage styles', () => {
  it('contains principle classes', () => {
    const { classesKeys } = testSetup();
    expect(habitsPageStyles).toBeDefined();
    classesKeys.forEach((keyValue) => {
      // @ts-ignore
      expect(habitsPageStyles[keyValue]).toBeDefined();
    });
  });
});

describe('HabitsPage decorators', () => {
  it('provides styled classes from decorators', () => {
    const { classesKeys } = testSetup();
    const component = mount(<HabitsPageDecorated />);
    // @ts-ignore
    const assignedClasses = component.find('HabitsPage').props().classes;
    classesKeys.forEach(keyValue => {
      expect(assignedClasses[keyValue]).toBeDefined();
    });
  });
});
