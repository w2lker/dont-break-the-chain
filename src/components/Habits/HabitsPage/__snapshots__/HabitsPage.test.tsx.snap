// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HabitsPage component match snapshot 1`] = `
"<div className=\\"habitsPage-wrapper\\">
  <WithStyles(HabitsHeader) />
</div>"
`;
